macos/FluentUI/Separator.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import AppKit
+
+@objc(MSFSeparatorStyle)
+public enum SeparatorStyle: Int {
+    case `default`
+    case shadow
```

Comment:
I initially made 2 styles to copy what we have on iOS. However, with the color choices I made (with input from design), there would be a significant different between 10.13 and 10.14. Either I find a better color than systemGray (I don't think I will), or I remove the "shadow" style altogether.

I'm more partial to removing the second style, considering that when I write the documentation, the first question will be "Why 2 styles, when is each used?" annnnnd I can't actually answer that 😅. For macOS we probably only need one style